src/java.base/macosx/classes/sun/nio/fs/BsdFileStore.java

```diff
@@ -77,4 +77,57 @@ UnixMountEntry findMountEntry() throws IOException {
 
         throw new IOException("Mount point not found in fstab");
     }
+
+    // returns true if extended attributes enabled on file system where given
+    // file resides, returns false if disabled or unable to determine.
+    private boolean isExtendedAttributesEnabled(UnixPath path) {
+        int fd = -1;
+        try {
+            fd = path.openForAttributeAccess(false);
+
+            // fgetxattr returns size if called with size==0
+            byte[] name = Util.toBytes("user.java");
+            BsdNativeDispatcher.fgetxattr(fd, name, 0L, 0);
+            return true;
+        } catch (UnixException e) {
+            // attribute does not exist
+            if (e.errno() == UnixConstants.ENOATTR)
+                return true;
+        } finally {
+            UnixNativeDispatcher.close(fd);
+        }
+        return false;
+    }
+
+    @Override
+    public boolean supportsFileAttributeView(Class<? extends FileAttributeView> type) {
+        // support DosFileAttributeView and UserDefinedAttributeView if extended
```

Comment:
A minor point here is that comment carrier over from the Linux implementation isn't quite right here as the BSD version doesn't support the "dos" attribute view.

Reply:
resolved in 5d00aa6